src/compute/cast/mod.rs

```diff
@@ -291,31 +324,50 @@ fn cast_large_to_list(array: &ListArray<i64>, to_type: &DataType) -> ListArray<i
 /// * Time32 and Time64: precision lost when going to higher interval
 /// * Timestamp and Date{32|64}: precision lost when going to higher interval
 /// * Temporal to/from backing primitive: zero-copy with data type change
-///
 /// Unsupported Casts
 /// * To or from `StructArray`
 /// * List to primitive
 /// * Utf8 to boolean
 /// * Interval and duration
 pub fn cast(array: &dyn Array, to_type: &DataType) -> Result<Box<dyn Array>> {
+    cast_with_options(array, to_type, CastOptions { wrapped: false })
+}
+
+/// Similar to cast, but overflowing cast will be wrapped
+/// Behavior:
+/// * PrimitiveArray to PrimitiveArray: overflowing cast will be wrapped (i.e. `256i16 as u8 = 0` vectorized).
+pub fn wrapped_cast(array: &dyn Array, to_type: &DataType) -> Result<Box<dyn Array>> {
+    cast_with_options(array, to_type, CastOptions { wrapped: true })
+}
+
+#[inline]
+pub fn cast_with_options(
```

Comment:
we can keep this private since `wrapped_cast` and `cast` cover all cases.

Reply:
But `dictionary_to_dictionary_values` is pub, which one do you prefer

1. Create `wrapping_dictionary_to_dictionary_values` and `wrapping_dictionary_to_dictionary_values`
2. Keep the CastOption public

Reply:
uhm, good point. I would make the two cases for now, just for consistency with Rust's std, but I agree that if we start having a lot of cases we may need to switch.

Is it a big pain for DataFuse to not have the option?

Reply:
Ok, both are good to me, I keep the CastOption as pub, it will not confuse users if we provide more apis.

If you'd like it to be private,  just add another version of `dictionary_to_dictionary_values ` and `dictionary_to_dictionary_keys`, this could be done in another pr.

Reply:
> Is it a big pain for DataFuse to not have the option?
Not yet, we just depend on the performance of `cast`. And I don't know if we will have more options in cast kernel.
As `SortOptions ` did.